File: app/lib/log_util.rb
# frozen_string_literal: true

class LogUtil
  def self.log_error(message, exception: nil, **details)
    # TODO(tiago): [CH-13826] add json support
    Rails.logger.error({
      message: message,
      exception: exception&.message,
      backtrace: exception&.backtrace,
      details: details,
    }.to_json)
    if exception
      # Exceptions have a default level of "error".
      Raven.capture_exception(
        exception,
        message: message,
        extra: details
      )
    end
  end

  # If you want to report a message rather than an exception you can use the log_message method.
  def self.log_message(message, **details)
    Raven.capture_message(
      message,
      level: "info",
      extra: details
    )
  end
end

File: app/assets/src/components/utils/logUtil.ts
import * as Sentry from "@sentry/browser";

// For more information on Sentry usage see: https://docs.sentry.io/platforms/javascript/usage/

interface LogErrorParams {
  message: string;
  exception?: Error;
  details?: { [key: string]: string };
}

export const logError = ({
  message,
  exception = null,
  details = {},
}: LogErrorParams) => {
  if (exception) {
    Sentry.captureException(exception, {
      extra: {
        message,
        details,
      },
    });
  } else {
    Sentry.captureMessage(message, {
      extra: details,
    });
  }
};
